### PR TITLE
[Docs] 2017 LAR, TS, and Panel field names and definitions

### LINKED ISSUE #826: Draft and Publish 2017 Field Names and Definitions
- [x] 2017 HMDA TS fields and definitions
- [x] 2017 HMDA LAR fields and definitions
- [x] 2017 HMDA Panel fields and definitions

Create text and pages that match [this style](https://ffiec.cfpb.gov/documentation/2018/panel-data-fields/) to provide stakeholders with documentation related to the 2017 HMDA collection.

### PR DESCRIPTION
**Pull request by meissadia**
Closes #826 

Screen Shot 2021-01-20 at 9 55 57 AM

### CODE REVIEW

File: src/documentation/markdown/2017/panel-data-fields.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+### [respondent\_rssd](#respondent_rssd)
+**Description:** The National Information Center RSSD of the institution
+
+**Values:**
+
+- -1: NULL/blank
+- Varying values
+
+### [respondent\_name](#respondent_name)
+**Description:** The name of the institution as registered with GLEIF
```
Comment (Kibrael) [inline src/documentation/markdown/2017/panel-data-fields.md:59]: for 2017 the institution name was not sourced from GLEIF, it came from the National Information Center or Transmittal Sheet if NIC did not have it.

File: src/documentation/markdown/2017/panel-data-fields.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+### [parent\_rssd](#parent_rssd)
+**Description:** The National Information Center's RSSD for the parent of the institution
+
+**Values:**
+
+- -1: NULL/blank
+- Varying values
+
+### [parent\_name](#parent_name)
+**Description:** The legal name of the parent institution
```
Comment (Kibrael) [inline src/documentation/markdown/2017/panel-data-fields.md:107]: sourced from the National Information Center (NIC)

File: src/documentation/markdown/2017/ts-data-fields.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+### [calendar\_quarter](#calendar_quarter)
+**Description:** The calandar quarter to which the data refer. Quarter 4 is for the annual submission of data.
+
+**Values:**
+- 4
+
+### [lei](#lei)
+**Description:** The Legal Entity Identifier (LEI) of the filing institution.
+
```
Comment (Kibrael) [inline src/documentation/markdown/2017/ts-data-fields.md:17]: LEI was not present in 2017.

Respondent ID was present in its place, [see this spec](https://s3.amazonaws.com/cfpb-hmda-public/prod/snapshot-data/2017_publicstatic_dataformat.pdf)

File: src/documentation/markdown/2017/ts-data-fields.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
+## Public Transmittal Sheet data fields with values and definitions
+
+### [activity\_year](#activity_year)
+**Description:** The calendar year of data collection.
+
+**Values:**
+- Integer values for year
+
+### [calendar\_quarter](#calendar_quarter)
```
Comment (Kibrael) [inline src/documentation/markdown/2017/ts-data-fields.md:9]: Calendar Quarter was not present in the 2017 schema.

File: src/documentation/markdown/2017/panel-data-fields.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+### [top\_holder\_rssd](#top_holder_rssd)
+**Description:** The National Information Center's RSSD for the top holder of the institution
+
+**Values:**
+
+- -1: NULL/blank
+- Varying values
+
+### [top\_holder\_name](#top_holder_name)
+**Description:** The legal name of the top holder of the institution
```
Comment (Kibrael) [inline src/documentation/markdown/2017/panel-data-fields.md:122]: sourced from the National Information Center (NIC)

File: src/documentation/markdown/2017/panel-data-fields.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+### [parent\_rssd](#parent_rssd)
+**Description:** The National Information Center's RSSD for the parent of the institution
+
+**Values:**
+
+- -1: NULL/blank
+- Varying values
+
+### [parent\_name](#parent_name)
```
Comment (Kibrael) [inline src/documentation/markdown/2017/panel-data-fields.md:106]: sourced from the National Information Center

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+- **Values:** 
+ - 1 - Single family with 1-4 units
+ - 2 - Manufactured housing
+ - 3 - Multifamily housing
+
+### [loan\_purpose](#loan_purpose)
+
+- **Description:** The purpose of covered loan or application
+- **Values:**
+  - 1 - Home purchase (one-to-four family)
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:48]: apologies, it looks like the PDF I referenced was incorrect for this field (I got the wrong year).

The correct enumerations are:
1.Home purchase
2.Home improvement
3.Refinancing

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 8]
```diff
@@ -0,0 +1,378 @@
...
+  - 2 - Home improvement (one-to-four family)
+  - 3 - Refinancing (home purchase or home improvement,
+ one-to-four family) 
+  - 4 - Multifamily dwelling (home purchase, home improvement, and refinancings)
+
+### [occupancy_type](#occupancy_type)
+
+- **Description:** Occupancy type for the dwelling
+- **Values:**
+  - 1 - Principal residence
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:58]: for 2017 these enumerations are:
1.Owner-occupied as a principal dwelling
2.Not owner-occupied
3.Not applicable

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 9]
```diff
@@ -0,0 +1,378 @@
...
+  - 2 - Second residence
+  - 3 - Investment property
+
+### [loan\_amount](#loan_amount)
+
+- **Description:** The amount of the covered loan, or the amount applied for
+- **Values:**
+  - Varying values  
+
+### [preapproval](#preapproval)
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:68]: for 2017: 
1.Preapproval was requested
2.Preapproval was not requested
3.Not applicable

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 10]
```diff
@@ -0,0 +1,378 @@
...
+
+- **Description:** Occupancy type for the dwelling
+- **Values:**
+  - 1 - Principal residence
+  - 2 - Second residence
+  - 3 - Investment property
+
+### [loan\_amount](#loan_amount)
+
+- **Description:** The amount of the covered loan, or the amount applied for
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:64]: rounded to the nearest thousand dollars.

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 11]
```diff
@@ -0,0 +1,378 @@
...
+- **Description:** Whether the covered loan or application involved a request for a preapproval of a home purchase loan under a preapproval program 
+- **Values:**
+  - 1 - Preapproval requested
+  - 2 - Preapproval not requested
+
+### [action\_taken](#action_taken)
+
+- **Description:** The action taken on the covered loan or application
+- **Values:**
+  - 1 - Loan originated
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:79]: for 2017: 

1.Loan originated
2.Application approved but not accepted
3.Application denied by financial institution
4.Application withdrawn by applicant
5.File closed for incompleteness
6.Loan purchased by your institution
7.Preapproval request denied by financial
institution
8.Preapproval request approved but not
accepted (optional reporting)

again, apologies, that must be a very old pdf.

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 12]
```diff
@@ -0,0 +1,378 @@
...
+  - 2 - Application approved but not accepted
+  - 3 - Application denied by financial institution 
+  - 4 - Application withdrawn by applicant
+  - 5 - File closed for incompleteness
+  - 6 - Loan purchased by your institution
+
+### [msa_md](#msa_md)
+
+- **Description:** The 5 digit MSA (metropolitan statistical area) or MD (metropolitan division) code. An MSA/MD is an area that has at least one urbanized area of 50,000 or more population.
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:88]: Note, that in 2017 MSA/MD values were reported by HMDA filers and not derived by the CFPB.

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 13]
```diff
@@ -0,0 +1,378 @@
...
+
+### [state\_code](#state_code)
+
+- **Description:** Two-letter state code
+- **Values:**
+  - Varying values
+
+### [county\_code](#county_code)
+
+- **Description:** State-county FIPS code
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:100]: five digit State-county FIPS code

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 14]
```diff
@@ -0,0 +1,378 @@
...
+- **Values:**
+  - Varying values
+
+### [census\_tract](#census_tract)
+
+- **Description:** 11 digit census tract number
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:106]: five digit Census Tract number with decimal. 
Example: 0109.02

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 15]
```diff
@@ -0,0 +1,378 @@
...
+- **Description:** Ethnicity of the applicant or borrower
+- **Values:**
+  - 1 - Hispanic or Latino
+  - 2 - Not Hispanic or Latino
+  - 3 - Information not provided by applicant in mail, internet, or telephone application
+  - 4 - Not applicable
+
+### [co\_applicant\_ethnicity\_1](#co_applicant_ethnicity_1)
+
+- **Description:** Ethnicity of the first co-applicant or co-borrower
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:121]: for 2017: 
1.Hispanic or Latino
2.Not Hispanic or Latino
3.Information not provided by applicant in
mail, Internet, or telephone application
(see App. A, I.D.2.)
4.Not applicable
5.No co-applicant

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 16]
```diff
@@ -0,0 +1,378 @@
...
+  - 4 - Hispanic
+  - 5 - White
+  - 6 - Information not provided by applicant in mail, internet, or telephone application
+  - 7 - Not applicable
+
+### [applicant\_race\_2](#applicant_race_2)
+
+- **Description:** Race of the applicant or borrower
+- **Values:**
+  - 1 - American Indian or Alaska Native
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:142]: for 2017: 
1.American Indian or Alaska Native
2.Asian
3.Black or African American
4.Native Hawaiian or Other Pacific Islander
5.White

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 17]
```diff
@@ -0,0 +1,378 @@
...
+  - 3 - Black or African American
+  - 4 - Hispanic
+  - 5 - White
+  - 6 - Information not provided by applicant in mail, internet, or telephone application
+  - 7 - Not applicable
+
+### [applicant\_race\_3](#applicant_race_3)
+
+- **Description:** Race of the applicant or borrower
+- **Values:**
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:153]: for 2017: 
1.American Indian or Alaska Native
2.Asian
3.Black or African American
4.Native Hawaiian or Other Pacific Islander
5.White

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 18]
```diff
@@ -0,0 +1,378 @@
...
+  - 6 - Information not provided by applicant in mail, internet, or telephone application
+  - 7 - Not applicable
+
+### [applicant\_race\_4](#applicant_race_4)
+
+- **Description:** Race of the applicant or borrower
+- **Values:**
+  - 1 - American Indian or Alaska Native
+  - 2 - Asian or Pacific Islander
+  - 3 - Black or African American
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:168]: for 2017: 
1.American Indian or Alaska Native
2.Asian
3.Black or African American
4.Native Hawaiian or Other Pacific Islander
5.White

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 19]
```diff
@@ -0,0 +1,378 @@
...
+  - 4 - Hispanic
+  - 5 - White
+  - 6 - Information not provided by applicant in mail, internet, or telephone application
+  - 7 - Not applicable
+
+### [applicant\_race\_5](#applicant_race_5)
+
+- **Description:** Race of the applicant or borrower
+- **Values:**
+  - 1 - American Indian or Alaska Native
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:178]: for 2017: 
1.American Indian or Alaska Native
2.Asian
3.Black or African American
4.Native Hawaiian or Other Pacific Islander
5.White

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 20]
```diff
@@ -0,0 +1,378 @@
...
+  - 2 - Asian or Pacific Islander
+  - 3 - Black or African American
+  - 4 - Hispanic
+  - 5 - White
+
+### [applicant\_sex](#applicant_sex)
+
+- **Description:** Sex of the applicant or borrower
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:241]: for 2017: 
1.Male
2.Female
3.Information not provided by applicant in
mail, Internet, or telephone application
(see App. A, I.D.2.)
4.Not applicable

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 21]
```diff
@@ -0,0 +1,378 @@
...
+  - 2 - Female
+  - 3 - Information not provided by applicant in mail, internet, or telephone application
+  - 4 - Not applicable
+  - 6 - Applicant selected both male and female
+
+### [co\_applicant\_sex](#co_applicant_sex)
+
+- **Description:** Sex of the first co-applicant or co-borrower
+- **Values:**
+  - 1 - Male
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:253]: for 2017:
1.Male
2.Female
3.Information not provided by applicant in
mail, Internet, or telephone application
(see App. A, I.D.2.)
4.Not applicable
5.No co-applicant

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 22]
```diff
@@ -0,0 +1,378 @@
...
+### [rate\_spread](#rate_spread)
+
+- **Description:** The difference between the covered loan’s annual percentage rate (APR) and the average prime offer rate (APOR) for a comparable transaction as of the date the interest rate is set
+- **Values:**
+  - Varying values
+
+### [hoepa\_status](#hoepa_status)
+
+- **Description:** Whether the covered loan is a high-cost mortgage
+- **Values:**
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:332]: for 2017: 
1.HOEPA loan
2.Not a HOEPA loan

File: src/documentation/markdown/2017/lar-data-fields.md [hunk 23]
```diff
@@ -0,0 +1,378 @@
...
+  - 1 - High-cost mortgage
+  - 2 - Not a high-cost mortgage
+  - 3 - Not applicable
+
+### [lien\_status](#lien_status)
+
+- **Description:** Lien status of the property securing the covered loan, or in the case of an application, proposed to secure the covered loan 
+- **Values:**
+  - 1 - Secured by a first lien
```
Comment (Kibrael) [inline src/documentation/markdown/2017/lar-data-fields.md:341]: for 2017:
1.Secured by a first lien
2.Secured by a subordinate lien
3.Not secured by a lien
4.Not applicable (purchased loans)

File: src/documentation/markdown/2017/panel-data-fields.md [hunk 24]
```diff
@@ -0,0 +1,126 @@
...
+
+### [activity\_year](#activity_year)
+**Description:** The collection year of the HMDA data to which the Panel file relates
+
+**Values:**
+
+- 2017
+
+### [lei](#lei)
+**Description:** A financial institution’s Legal Entity Identifier
```
Comment (Kibrael) [inline src/documentation/markdown/2017/panel-data-fields.md:11]: LEI not present in 2017

File: src/documentation/markdown/2017/panel-data-fields.md [hunk 25]
```diff
@@ -0,0 +1,126 @@
...
+
+**Values:**
+
+- Varying values
+
+### [tax\_id](#tax_id)
+**Description:** The federal tax ID of the institution
```
Comment (Kibrael) [inline src/documentation/markdown/2017/panel-data-fields.md:18]: in 10-digit format with a dash.

File: src/documentation/markdown/2017/panel-data-fields.md [hunk 26]
```diff
@@ -0,0 +1,126 @@
...
+- Varying values
+
+### [agency\_code](#agency_code)
+**Description:** The integer code corresponding to an institution's regulatory agency
+
+**Values:**
+  - 1: Office of the Comptroller of the Currency (OCC)
+  - 2: Federal Reserve System (FRB)
+  - 3: Federal Deposit Insurance Corporation (FDIC)
+  - 4: Office of Thrift Supervision (OTS)
```
Comment (Kibrael) [inline src/documentation/markdown/2017/panel-data-fields.md:31]: remove 4, it's not current

File: src/documentation/markdown/2017/panel-data-fields.md [hunk 27]
```diff
@@ -0,0 +1,126 @@
...
+  - 5: National Credit Union Administration (NCUA)
+  - 7: Department of Housing and Urban Development (HUD)
+  - 9: Consumer Financial Protection Bureau (CFPB)
+
+### [id\_2017](#id_2017)
+**Description:** The 2017 HMDA Platform primary identifier for an institution. This identifier is used in naming modified LAR files on [this page](https://ffiec.cfpb.gov/data-publication/modified-lar/2017)
```
Comment (Kibrael) [inline src/documentation/markdown/2017/panel-data-fields.md:37]: remove the first sentence to reduce confusion for users.

File: src/documentation/markdown/2017/panel-data-fields.md [hunk 28]
```diff
@@ -0,0 +1,126 @@
...
+
+**Values:**
+
+- Varying values
+
+### [arid\_2017](#arid_2017)
+**Description:** The concatenation of an institution's 2017 Agency Code and Respondent ID. In order to match between 2017 and 2018, take the arid_2017 column from the 2018 panel and use this to join to the concatenation of the agency code and respondent ID in the 2017 panel. Respondent ID in 2017 and prior is not a unique value and must be joined to agency code to generate a primary key for the HMDA dataset. For 2018 and forward, LEI will be the primary key.
```
Comment (Kibrael) [inline src/documentation/markdown/2017/panel-data-fields.md:44]: please add the following:

for more information on institution identifiers, please see: https://ffiec.cfpb.gov/documentation/2020/identifiers-faq/

File: src/documentation/markdown/2017/lar-data-fields.md (hunk 14 above)
Comment (meissadia) [reply src/documentation/markdown/2017/lar-data-fields.md:106]: Removing the leading zero from the example to make it 5 digits + decimal.

Comment (Kibrael) [reply src/documentation/markdown/2017/lar-data-fields.md]: oh, nice catch!

**Comment by Kibrael** [comment]
This looks great, thank you so much for putting this together!

**Review by Kibrael — APPROVED** [review]
Great stuff!

**Pull request merged**